src/js/users/components/__tests__/Create.test.jsx

```diff
@@ -22,80 +51,46 @@ describe("<CreateUser />", () => {
             handle: ""
         };
+        history = createBrowserHistory();
     });
 
-    it("should render", () => {
-        const wrapper = shallow(<CreateUser {...props} />);
-        expect(wrapper).toMatchSnapshot();
+    it("should render correctly when show=true", () => {
+        renderWithRouter(<CreateUser {...props} />, state, history);
+        expect(screen.getByText("Create User")).toBeInTheDocument();
+        expect(screen.getByText("Username")).toBeInTheDocument();
+        expect(screen.getByText("Password")).toBeInTheDocument();
+        expect(screen.getByText("Force user to reset password on login"));
+        expect(screen.getByRole("button", { name: "Save" })).toBeInTheDocument();
     });
 
-    it("should render with error", () => {
-        const wrapper = shallow(<CreateUser {...props} />);
-        wrapper.setState({
-            errorPassword: "Password too short"
-        });
-        expect(wrapper).toMatchSnapshot();
+    it("should render correct error messages", () => {
+        renderWithRouter(<CreateUser {...props} />, state, history);
+        const saveButton = screen.getByRole("button", { name: "Save" });
+        saveButton.click();
+        expect(screen.getByText("Please specify a username")).toBeInTheDocument();
+        expect(screen.getByText("Passwords must contain at least 8 characters")).toBeInTheDocument();
+        expect(props.onCreate).not.toHaveBeenCalled();
     });
 
-    it("should render when name has changed", () => {
-        const e = {
-            target: { name: "handle", value: "bob" }
-        };
-        const wrapper = shallow(<CreateUser {...props} error="Error" />);
-        expect(wrapper).toMatchSnapshot();
-        wrapper.find(Input).simulate("change", e);
-        expect(props.onClearError).toHaveBeenCalled();
-        expect(wrapper).toMatchSnapshot();
-    });
-
-    it("should render when password has changed", () => {
-        const e = {
-            target: { name: "password", value: "password" }
-        };
-        const wrapper = shallow(<CreateUser {...props} error="Error" />);
-        expect(wrapper).toMatchSnapshot();
-        wrapper.find(PasswordInput).simulate("change", e);
-        expect(props.onClearError).toHaveBeenCalled();
-        expect(wrapper).toMatchSnapshot();
+    it("should not display error messages and should call onCreate when valid info entered", () => {
+        renderWithRouter(<CreateUser {...props} />, state, history);
+        const userNameInput = screen.getByLabelText("username");
+        const passwordInput = screen.getByLabelText("password");
+        const saveButton = screen.getByRole("button", { name: "Save" });
+        fireEvent.change(userNameInput, { target: { value: "newUsername" } });
+        fireEvent.change(passwordInput, { target: { value: "newPassword" } });
+        expect(props.onCreate).not.toHaveBeenCalled();
+        saveButton.click();
+        expect(props.onCreate).toHaveBeenCalled();
```

Comment:
Example of making test more readable.

```suggestion
        renderWithRouter(<CreateUser {...props} />, state, history);
        
        fireEvent.change(screen.getByLabelText("username"), { target: { value: "newUsername" } });
        fireEvent.change(screen.getByLabelText("password"), { target: { value: "newPassword" } });
        
        expect(props.onCreate).not.toHaveBeenCalled();
        
        screen.getByRole("button", { name: "Save" }).click();
        
        expect(props.onCreate).toHaveBeenCalled();
```